tooling/create-tauri-app/src/recipes/solid.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019-2021 Tauri Programme within The Commons Conservancy
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-License-Identifier: MIT
+
+import { join } from 'path'
+import { shell } from '../shell'
+import { Recipe } from '../types/recipe'
+
+const solid: Recipe = {
+  descriptiveName: {
+    name: 'Solid (https://github.com/solidjs/solid)',
+    value: 'solid'
+  },
+  shortName: 'solid',
+  extraNpmDevDependencies: [],
+  extraNpmDependencies: [],
+  extraQuestions: ({ ci }) => {
+    return [
+      {
+        type: 'list',
+        name: 'template',
+        message: 'Which Solid template would you like to use? (Read more at https://github.com/solidjs/templates)',
+        choices: [
+          'js',
+          'ts-bootstrap',
+          'ts-minimal',
+          'ts-router',
+          'ts-windicss',
+          'ts'
+        ],
+        default: 'ts',
+        loop: false,
+        when: !ci
+      }
+    ]
+  },
+  configUpdate: ({ cfg, packageManager }) => ({
+    ...cfg,
+    distDir: `../public`,
+    devPath: 'http://localhost:3000',
+    beforeDevCommand: `${
+      packageManager === 'npm' ? 'npm run' : packageManager
+    } dev`,
+    beforeBuildCommand: `${
+      packageManager === 'npm' ? 'npm run' : packageManager
+    } build`
+  }),
+  preInit: async ({ cwd, cfg, answers }) => {
+    await shell(
+      'npx',
+      ['degit', `solidjs/templates/${answers?.template}`, `${cfg.appName}`],
```

Comment:
```suggestion
      ['degit', `solidjs/templates/${answers?.template ?? 'js'}`, cfg.appName],
```
This will ensure that there is a template when `ci` mode is on, otherwise the resulting string would be `'solidjs/templates/undefined'`